### PR TITLE
Fix: Exception when parsing log #20966

### LINKED ISSUE #20966: Exception when parsing log
### Apache Airflow version

2.1.4

### What happened

[2022-01-19 13:42:46,107] {taskinstance.py:1463} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1165, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1283, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 1313, in _execute_task
    result = task_copy.execute(context=context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py", line 367, in execute
    final_state, remote_pod, result = self.create_new_pod_for_operator(labels, launcher)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py", line 521, in create_new_pod_for_operator
    final_state, remote_pod, result = launcher.monitor_pod(pod=self.pod, get_logs=self.get_logs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/cncf/kubernetes/utils/pod_launcher.py", line 148, in monitor_pod
    timestamp, message = self.parse_log_line(line.decode('utf-8'))
UnicodeDecodeError: 'utf-8' codec can't decode bytes in position 16413-16414: invalid continuation byte

### What you expected to happen

_No response_

### How to reproduce

_No response_

### Operating System

Debian GNU/Linux 10 (buster)

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon==2.2.0
apache-airflow-providers-celery==2.0.0
apache-airflow-providers-cncf-kubernetes==2.0.2
apache-airflow-providers-docker==2.1.1
apache-airflow-providers-elasticsearch==2.0.3
apache-airflow-providers-ftp==2.0.1
apache-airflow-providers-google==5.1.0
apache-airflow-providers-grpc==2.0.1
apache-airflow-providers-hashicorp==2.1.0
apache-airflow-providers-http==2.0.1
apache-airflow-providers-imap==2.0.1
apache-airflow-providers-microsoft-azure==3.1.1
apache-airflow-providers-mysql==2.1.1
apache-airflow-providers-postgres==2.2.0
apache-airflow-providers-redis==2.0.1
apache-airflow-providers-sendgrid==2.0.1
apache-airflow-providers-sftp==2.1.1
apache-airflow-providers-slack==4.0.1
apache-airflow-providers-sqlite==2.0.1
apache-airflow-providers-ssh==2.1.1

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

_No response_

### Anything else

Always on a specific docker container.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by jakubno**
Took after #21053 due to inactivity

>When the task outputs some strange characters, decoding will lead to "UnicodeDecodeError" error due to inconsistent encoding. We can try-catch it or ignore it by specifying the parameter: errors="backslashreplace" .
e.g.

            line = ''
            for raw_line in iter(self.sub_process.stdout.readline, b''):
                try:
                    line = raw_line.decode(output_encoding).rstrip()
                except UnicodeDecodeError as err:
                    print(err, output_encoding, raw_line)
                self.log.info("%s", line)
>closes: https://github.com/apache/airflow/issues/20966

### CODE REVIEW

File: tests/hooks/test_subprocess.py [hunk 1]
```diff
@@ -96,3 +96,9 @@ def test_should_exec_subprocess(self, mock_popen, mock_temporary_directory):
             stderr=STDOUT,
             stdout=PIPE,
         )
+
+    def test_task_decode(self):
+        hook = SubprocessHook()
+        command=['bash', '-c', "source "+__file__[:-2]+'sh']
+        result = hook.run_command(command=command)
+        assert result.exit_code==0
```
Comment (uranusjr) [inline tests/hooks/test_subprocess.py:104]: What does this test?

Comment (uranusjr) [inline tests/hooks/test_subprocess.py]: Instead of needing a separate script, I think you can just use `sys.executable` to print some weird characters.

Comment (jakubno) [reply tests/hooks/test_subprocess.py]: Simplified the test, so the script isn't needed anymore

**Review by eladkal — APPROVED** [review]

**Comment by eladkal** [comment]
This PR is partially providers and partially core so I'm setting also 2.3.1 milestone
@potiuk will it make it on time to the providers wave today?

**Comment by potiuk** [comment]
> This PR is partially providers and partially core so I'm setting also 2.3.1 milestone @potiuk will it make it on time to the providers wave today?

I am waiting for few more comments on few PRs so - yeah possible.

**Review by potiuk — APPROVED** [review]

**Review by uranusjr — APPROVED** [review]

**Pull request merged**